userspace/libsinsp/parsers.cpp

```diff
@@ -632,19 +632,11 @@ bool sinsp_parser::reset(sinsp_evt *evt) {
 		//
 		// Error detection logic
 		//
-		if(evt->get_num_params() != 0 && ((evt->get_info()->params[0].name[0] == 'r' &&
-		                                   evt->get_info()->params[0].name[1] == 'e' &&
-		                                   evt->get_info()->params[0].name[2] == 's' &&
-		                                   evt->get_info()->params[0].name[3] == '\0') ||
-		                                  (evt->get_info()->params[0].name[0] == 'f' &&
-		                                   evt->get_info()->params[0].name[1] == 'd' &&
-		                                   evt->get_info()->params[0].name[2] == '\0'))) {
-			if(evt->has_return_value()) {
-				int64_t res = evt->get_syscall_return_value();
-
-				if(res < 0) {
-					evt->set_errorcode(-(int32_t)res);
-				}
+		if(evt->has_return_value()) {
```

Comment:
if i don't miss something all the syscall exit event should have at least the return value so i don't see why we need the additional manual check on the parameter names

Reply:
I encountered this code right yesterday and was wondering exactly the same :rofl:

Reply:
BTW this logic was broken since there are all these possible values `"uid", "gid", "res_or_fd", "euid", "egid"`